Compiling geangy-plugins appeared errors

I  use msys2 to compile geany-plugin-1.38 source code ;when I excute "make j2" in msys2 ;it shows:
"collect2.exe:error :ld return 1 exit status " "make[2]: *** [makefile:762:viw.exe] Error 1";
I want to know how to solve it

> @rdipardo probably not, the problems are in linking, not in compiling.

Headers can provide `extern` declarations, which compile but still confound the linker, e.g.

~~~c
/*
 * undef.h
 */
extern int foo(void);
~~~

~~~c
/*
 * undef.c
 */
#include "undef.h"
int main(void)
{
   return foo();
}
~~~

~~~sh-session
$ gcc -Wall -Wextra -c undef.c
$ echo $?
0
$ gcc undef.o
/usr/bin/ld: undef.o: in function `main':
undef.c:(.text+0x5): undefined reference to `foo'
collect2: error: ld returned 1 exit status
~~~

Replacing core type definitions with `extern` declarations in dependent headers would be a pretty nasty trick on GCC's part, but not impossible.

--- a/vimode/src/Makefile.am
+++ b/vimode/src/Makefile.am
@@ -3,8 +3,6 @@ plugin = vimode
 
 geanyplugins_LTLIBRARIES = vimode.la
 
-noinst_PROGRAMS = viw
-
 vi_srcs = \
 	vi.h \
 	vi.c \
@@ -42,19 +40,23 @@ vi_srcs = \
 vimode_la_SOURCES = \
 	backends/backend-geany.c \
 	$(vi_srcs)
-
-viw_SOURCES = \
-	backends/backend-viw.c \
-	$(vi_srcs)
     
 vimode_la_CPPFLAGS = $(AM_CPPFLAGS) \
 	-DG_LOG_DOMAIN=\"Vimode\"
 vimode_la_CFLAGS = $(AM_CFLAGS)
 vimode_la_LIBADD = $(COMMONLIBS)
 
+if !MINGW
+noinst_PROGRAMS = viw
+
+viw_SOURCES = \
+	backends/backend-viw.c \
+	$(vi_srcs)
+
 viw_CPPFLAGS = $(AM_CPPFLAGS) \
 	-DG_LOG_DOMAIN=\"Vimode\"
 viw_CFLAGS = $(AM_CFLAGS)
 viw_LDADD = $(COMMONLIBS)
+endif
 
 include $(top_srcdir)/build/cppcheck.mk

